controllers/costmanagement_controller.go

```diff
@@ -147,13 +150,28 @@ func ReflectSpec(r *CostManagementReconciler, cost *costmgmtv1alpha1.CostManagem
 		costConfig.UploadCycle = costmgmtv1alpha1.DefaultUploadCycle
 	}
 
-	costConfig.SourceName = StringReflectSpec(r, cost, &cost.Spec.Source.SourceName, &cost.Status.Source.SourceName, "")
+	var sourceNameChanged bool
+	costConfig.SourcesAPIPath, _ = StringReflectSpec(r, cost, &cost.Spec.Source.SourcesAPIPath, &cost.Status.Source.SourcesAPIPath, costmgmtv1alpha1.DefaultSourcesPath)
+	costConfig.SourceName, sourceNameChanged = StringReflectSpec(r, cost, &cost.Spec.Source.SourceName, &cost.Status.Source.SourceName, "")
 	costConfig.CreateSource = false
 	if cost.Spec.Source.CreateSource != nil {
 		costConfig.CreateSource = *cost.Spec.Source.CreateSource
 	}
+	sourceCycleChange := false
+	if !reflect.DeepEqual(cost.Spec.Source.CheckCycle, cost.Status.Source.CheckCycle) {
+		cost.Status.Source.CheckCycle = cost.Spec.Source.CheckCycle
+		sourceCycleChange = true
+	}
+	if cost.Status.Source.CheckCycle != nil {
+		costConfig.SourceCheckCycle = *cost.Status.Source.CheckCycle
+	} else {
+		costConfig.SourceCheckCycle = costmgmtv1alpha1.DefaultSourceCheckCycle
+	}
+	if !sourceNameChanged || !sourceCycleChange {
```

Comment:
```suggestion
	if !sourceNameChanged && !sourceCycleChange {
```
This needs to be an `and`, not an `or`.